spec/ddtrace/configuration/settings_spec.rb

```diff
@@ -6,6 +6,178 @@
 RSpec.describe Datadog::Configuration::Settings do
   subject(:settings) { described_class.new }
 
+  describe '#analytics' do
```

Comment:
Backfilled a lot of tests for configuration settings in this file that was previously missing.